tasks/db_deploy/test/unit_tests/test_migrate_db.py

```diff
@@ -320,3 +253,92 @@ def test_migrate_versions_1_to_2_happy_path(
+    @patch("migrate_db.collections_table_sql")
+    @patch("migrate_db.provider_collection_xref_table_sql")
+    @patch("migrate_db.granules_table_sql")
+    @patch("migrate_db.files_table_sql")
+    @patch("migrate_db.get_admin_connection")
+    @patch("migrate_db.text")
+    def test_migrate_versions_3_to_4_happy_path(
+        self,
+        mock_text: MagicMock,
+        mock_connection: MagicMock,
+        mock_files_table: MagicMock,
+        mock_granules_table: MagicMock,
+        mock_provider_collection_xref_table: MagicMock,
+        mock_collections_table: MagicMock,
+        mock_providers_table: MagicMock,
+        mock_schema_versions_data: MagicMock,
+    ):
+        """
+        Tests the migrate_versions_3_to_4 function happy path
+        """
+        for latest_version in [True, False]:
+            with self.subTest(latest_version=latest_version):
+                # Setup the mock object that conn.execute is a part of in
+                # the connection with block
+                mock_conn_enter = mock_connection().connect().__enter__()
+
+                # Run the function
+                migrate_db.migrate_versions_3_to_4(self.config, latest_version)
+
+                # Check that all of the functions were called the correct
+                # number of times with the proper values
+                mock_connection.assert_any_call(
+                    self.config, self.config["user_database"]
+                )
+                mock_providers_table.assert_called_once()
+                mock_collections_table.assert_called_once()
+                mock_provider_collection_xref_table.assert_called_once()
+                mock_granules_table.assert_called_once()
+                mock_files_table.assert_called_once()
+
+                # Check the text calls occur and in the proper order
+                text_calls = [
+                    call("SET ROLE orca_dbo;"),
+                    call("SET search_path TO orca, public;"),
+                ]
+                mock_text.assert_has_calls(text_calls, any_order=False)
+
+                # Validate logic switch and set the execution order
+                if latest_version:
+                    mock_schema_versions_data.assert_called_once()
+                    execution_order = [
+                        call.execute(mock_text("SET ROLE orca_dbo;")),
+                        call.execute(mock_text("SET search_path TO orca, public;")),
+                        call.execute(mock_providers_table()),
+                        call.execute(mock_collections_table()),
+                        call.execute(mock_provider_collection_xref_table()),
+                        call.execute(mock_granules_table()),
+                        call.execute(mock_files_table()),
+                        call.commit(),
+                    ]
+
+                else:
+                    mock_schema_versions_data.assert_not_called()
+                    execution_order = [
```

Comment:
It looks like `execution_order` is the same in both these blocks. If that is the case, I would pull it out of the blocks and only define it once.

Reply:
yes removed duplicate